localstack-core/localstack/services/cloudwatch/provider_v2.py

```diff
@@ -835,6 +831,91 @@ def _get_timestamp(input: dict):
             history = [h for h in history if (date := _get_timestamp(h)) and date <= end_date]
         return DescribeAlarmHistoryOutput(AlarmHistoryItems=history)
 
+    def _evaluate_composite_alarms(self, context: RequestContext, triggering_alarm):
+        store = self.get_store(context.account_id, context.region)
+        alarms = list(store.alarms.values())
+        composite_alarms = [a for a in alarms if isinstance(a, LocalStackCompositeAlarm)]
+        for composite_alarm in composite_alarms:
+            self._evaluate_composite_alarm(context, composite_alarm, triggering_alarm)
+
+    def _evaluate_composite_alarm(self, context, composite_alarm, triggering_alarm):
+        store = self.get_store(context.account_id, context.region)
+        new_state_value = StateValue.OK
+        alarm_rule = composite_alarm.alarm["AlarmRule"]
+        # assuming that a rule consists only of ALARM evaluations of metric alarms, with OR logic applied
+        metric_alarm_arns = re.findall(
+            r'\("([^"]*)"\)', alarm_rule
+        )  # regexp for everything within (" ")
+        for metric_alarm_arn in metric_alarm_arns:
+            metric_alarm = store.alarms.get(metric_alarm_arn)
+            if metric_alarm.alarm["StateValue"] == StateValue.ALARM:
+                triggering_alarm = metric_alarm
+                new_state_value = StateValue.ALARM
+                break
+        old_state_value = composite_alarm.alarm["StateValue"]
+        if old_state_value == new_state_value:
+            return
+        triggering_alarm_arn = triggering_alarm.alarm.get("AlarmArn")
+        triggering_alarm_state = triggering_alarm.alarm.get("StateValue")
+        triggering_alarm_state_change_timestamp = triggering_alarm.alarm.get(
+            "StateTransitionedTimestamp"
+        )
+        state_reason_formatted_timestamp = triggering_alarm_state_change_timestamp.strftime(
+            "%A %d %B, %Y %H:%M:%S %Z"
+        )
+        state_reason = (
+            f"{triggering_alarm_arn} "
+            f"transitioned to {triggering_alarm_state} "
+            f"at {state_reason_formatted_timestamp}"
+        )
+        state_reason_data = {
+            "triggeringAlarms": [
+                {
+                    "arn": triggering_alarm_arn,
+                    "state": {
+                        "value": triggering_alarm_state,
+                        "timestamp": timestamp_millis(triggering_alarm_state_change_timestamp),
+                    },
+                }
+            ]
+        }
+        self._update_state(
+            context, composite_alarm, new_state_value, state_reason, state_reason_data
+        )
+        if composite_alarm.alarm["ActionsEnabled"]:
+            self._run_composite_alarm_actions(
+                context, composite_alarm, old_state_value, triggering_alarm
+            )
+
+    def _run_composite_alarm_actions(
```

Comment:
There is some duplication between this method and [the existing step in the `set_alarm_state` method](https://github.com/localstack/localstack/blob/385ed1508cfa91b6cc4859f89b57ec7385cff765/localstack-core/localstack/services/cloudwatch/provider_v2.py#L359). Do you think it's worth combining the implementations?

Reply:
I agree with your assessment, I just wanted to raise the question.